ui-components/src/helpers/tableSummaries.tsx

```diff
@@ -1,71 +1,91 @@
 import * as React from "react"
 import { t } from "./translator"
-import { UnitSummary } from "@bloom-housing/backend-core/types"
+import { UnitSummary, UnitsSummary } from "@bloom-housing/backend-core/types"
 import { GroupedTableGroup } from "../tables/GroupedTable"
 
-export const unitSummariesTable = (summaries: UnitSummary[]) => {
-  const unitSummaries = summaries.map((unitSummary) => {
-    const minIncome =
-      unitSummary.minIncomeRange.min == unitSummary.minIncomeRange.max ? (
-        <strong>{unitSummary.minIncomeRange.min}</strong>
-      ) : (
-        <>
-          <strong>{unitSummary.minIncomeRange.min}</strong> {t("t.to")}{" "}
-          <strong>{unitSummary.minIncomeRange.max}</strong>
-        </>
-      )
+export const getSummaryRow = (
+  minIncomeRangeMin: string | undefined,
+  minIncomeRangeMax: string | undefined,
+  rentRangeMin: string | undefined,
+  rentRangeMax: string | undefined,
+  rentAsPercentIncomeRangeMin: string | number,
+  rentAsPercentIncomeRangeMax: string | number,
+  totalAvailable: number | undefined,
+  totalCount: number | undefined,
+  unitTypeName: string
+) => {
+  const minIncome =
+    minIncomeRangeMin == minIncomeRangeMax ? (
+      <strong>{minIncomeRangeMin}</strong>
+    ) : (
+      <>
+        <strong>{minIncomeRangeMin}</strong> {t("t.to")} <strong>{minIncomeRangeMax}</strong>
+      </>
+    )
 
-    const getRent = (rentMin: string, rentMax: string, percent = false) => {
-      const unit = percent ? `% ${t("t.income")}` : ` ${t("t.perMonth")}`
-      return rentMin == rentMax ? (
-        <>
-          <strong>{rentMin}</strong>
-          {unit}
-        </>
-      ) : (
-        <>
-          <strong>{rentMin}</strong> {t("t.to")} <strong>{rentMax}</strong>
-          {unit}
-        </>
-      )
-    }
+  const getRent = (rentMin: string | undefined, rentMax: string | undefined, percent = false) => {
```

Comment:
Instead of `string | undefined` here, it would be more readable to have the type as just `string`, and in the body of the function return an empty string early if both are undefined

Reply:
updated